narwhal/worker/src/batch_maker.rs

```diff
@@ -211,5 +268,60 @@ impl BatchMaker {
             .created_batch_latency
             .with_label_values(&[self.committee.epoch.to_string().as_str(), reason])
             .observe(self.batch_start_timestamp.elapsed().as_secs_f64());
+
+        // Clone things to not capture self
+        let store = self.store.clone();
+        let worker_id = self.id;
+        let tx_digest = self.tx_digest.clone();
+        let metadata = batch.metadata.clone();
+
+        async move {
+            // The channels are closed, so we are shutting down. Nothing to do.
+            if closing {
+                return;
+            }
```

Comment:
An alternative it would be instead of doing this here to make the `seal` method return an `Option` instead (ex `Option<impl Future<Output = ()>>`) .If the result is `Some` then can be included to the futures list, otherwise it ignores. That will allow you to return directly in line 261 (`return None;`) and avoid handling the `closing` variable within the future

Reply:
Oh I like that!